src/linear_algebra/quadratic_form.lean

```diff
@@ -788,3 +789,189 @@ theorem exists_orthogonal_basis [hK : invertible (2 : K)]
+  rw [sum_right, sum_eq_single j],
+  { rw [smul_left, smul_right], ring },
+  { intros i _ hij,
+    rw [smul_left, smul_right,
+        show (associated_hom R₁) Q (v j) (v i) = 0, by exact hv₂ i j hij,
+        mul_zero, mul_zero] },
+  { intro hnj, exact false.elim (hnj hj) }
+end
+
+/-- The weighted sum of squared with respect some weight. `weighted_sum_squares` is the
+quadratic form version of this. -/
+def weighted_sum_squares' (w : ι → R) : (ι → R) → R :=
+  λ x, ∑ i : ι, w i * x i * x i
+
+@[simp]
+lemma weighted_sum_squares'_apply (w v : ι → R) :
+  weighted_sum_squares' w v = ∑ i : ι, w i * v i * v i := rfl
+
+lemma weighted_sum_squares'_smul {w : ι → R₁} (a : R₁) (x : ι → R₁) :
+  weighted_sum_squares' w (a • x) = a * a * weighted_sum_squares' w x :=
+begin
+  simp only [weighted_sum_squares', algebra.id.smul_eq_mul, pi.smul_apply],
+  conv_rhs { rw [mul_assoc, mul_sum, mul_sum] },
+  refine sum_congr rfl (λ _ _, by ring),
+end
+
+lemma weighted_sum_squares'_polar_add_left {w : ι → R₁} (x x' y : ι → R₁) :
+  polar (weighted_sum_squares' w) (x + x') y =
+  polar (weighted_sum_squares' w) x y + polar (weighted_sum_squares' w) x' y :=
+begin
+  simp_rw [weighted_sum_squares', polar, pi.add_apply],
+  iterate 7 { rw ← sum_add_distrib <|> rw ← sum_sub_distrib },
+  refine sum_congr rfl (λ _ _, by ring),
+end
+
+lemma weighted_sum_squares'_polar_smul_left {w : ι → R₁} (a : R₁) (x y : ι → R₁) :
+  polar (weighted_sum_squares' w) (a • x) y = a • polar (weighted_sum_squares' w) x y :=
+begin
+  simp_rw [weighted_sum_squares', polar, pi.add_apply, pi.smul_apply],
+  iterate 4 { rw ← sum_add_distrib <|> rw ← sum_sub_distrib },
+  simp_rw [smul_sum, smul_eq_mul],
+  refine sum_congr rfl (λ _ _, by ring),
+end
+
+lemma weighted_sum_squares'_polar_add_right {w : ι → R₁} (x y y' : ι → R₁) :
+  polar (weighted_sum_squares' w) x (y + y') =
+  polar (weighted_sum_squares' w) x y + polar (weighted_sum_squares' w) x y' :=
+begin
+  simp_rw [weighted_sum_squares', polar, pi.add_apply],
+  iterate 7 { rw ← sum_add_distrib <|> rw ← sum_sub_distrib },
+  refine sum_congr rfl (λ _ _, by ring),
+end
+
+lemma weighted_sum_squares'_polar_smul_right {w : ι → R₁} (a : R₁) (x y : ι → R₁) :
+  polar (weighted_sum_squares' w) x (a • y) = a • polar (weighted_sum_squares' w) x y :=
+begin
+  simp_rw [weighted_sum_squares', polar, pi.add_apply, pi.smul_apply],
+  iterate 4 { rw ← sum_add_distrib <|> rw ← sum_sub_distrib },
+  simp_rw [smul_sum, smul_eq_mul],
+  refine sum_congr rfl (λ _ _, by ring),
+end
+
+/-- `weighted_sum_squares'` represented as a quadratic form. -/
+def weighted_sum_squares (w : ι → R₁) : quadratic_form R₁ (ι → R₁) :=
+{ to_fun := weighted_sum_squares' w,
+  to_fun_smul := weighted_sum_squares'_smul,
+  polar_add_left' := weighted_sum_squares'_polar_add_left,
+  polar_smul_left' := weighted_sum_squares'_polar_smul_left,
+  polar_add_right' := weighted_sum_squares'_polar_add_right,
+  polar_smul_right' := weighted_sum_squares'_polar_smul_right }
```

Comment:
I think you can build this far more succinctly as
```suggestion
/-- `weighted_sum_squares'` represented as a quadratic form. -/
def weighted_sum_squares (w : ι → R₁) : quadratic_form R₁ (ι → R₁) :=
∑ i : ι, w i • quadratic_form.proj i i
```

Reply:
The proof of `_apply` becomes:
```

@[simp]
lemma weighted_sum_squares_apply (w v : ι → R₁) :
  weighted_sum_squares w v = ∑ i : ι, w i * (v i * v i) :=
quadratic_form.sum_apply _ _ _
```

Reply:
Thanks! This is very nice. I can't seem to find `quadratic_form.sum_apply`. Which file is it in?

Reply:
This one: #7417